test/XmlHttpRequestConnection.tests.js

```diff
@@ -145,6 +145,36 @@ define(function (require) {
 
         });
 
+        describe("is correctly using XmlHttpRequest and runs attached callbacks while performing:", function (){
+            beforeEach(function (){
+                mockXMLHttpRequest.prototype.send = function (body) {
+                    this.readyState = 4;
+                    this.status = 200;
+                    this.onloadstart();
+                    this.onreadystatechange();
+                };
+                spyOn(mockXMLHttpRequest.prototype, 'send').andCallThrough();
+
+                window.XMLHttpRequest = (function () { return mockXMLHttpRequest; }());
+
+                connection = new XmlHttpRequestConnection();
+            });
+
+            it("execute call and invoke onloadstart callback", function () {
```

Comment:
I'm not sure how should I test it. If I run them manually inside the `mockXMLHttpRequest.prototype.send` mock then it is not something we can rely on.
I'm open for any suggestions in that topic @dpobel